pkg/common/cluster_state.go

```diff
@@ -55,6 +65,7 @@ type ClusterState struct {
 	PostgresqlServiceEndpoints      *v1.Endpoints
 	PodDisruptionBudget             *v1beta12.PodDisruptionBudget
 	KeycloakProbes                  *v1.ConfigMap
+	KeycloakBackup                  *v1alpha1.KeycloakBackup
```

Comment:
What will happen if you will be migrating multiple times, e.g. Keycloak 10 -> 11 -> 12? Maybe this should be a list and the Operator should always consider it's last element?

Reply:
My idea is
Every time Keycloak gets migration, a OneTimeBackup will be done before the upgrade.
There is the possible scenario:
1st migration(10->11) -> 1st Backup -> 1st Keycloak upgrade -> 2nd migration(11->12) -> 2nd Backup -> 2nd Keycloak upgrade
---
Even two migrations can be executed in a short time. Duplicated backup won't create any error because they are named by time (e.g. "migrate-backup-20200608-163918").

Reply:
Ok, this makes sense.